crates/swc_ecma_lints/src/rules/no_alert.rs

```diff
@@ -0,0 +1,158 @@
+}
+
+impl NoAlert {
+    fn new(
+        expected_reaction: LintRuleReaction,
+        top_level_declared_vars: AHashSet<Id>,
+        top_level_ctxt: SyntaxContext,
+        es_version: EsVersion,
+    ) -> Self {
+        Self {
+            expected_reaction,
+            top_level_ctxt,
+            top_level_declared_vars,
+            pass_call_on_global_this: es_version < EsVersion::Es2020,
+            call_expr_span: None,
+        }
+    }
+
+    fn emit_report(&self, fn_name: &str) {
+        let span = self.call_expr_span.unwrap();
+        let message = format!("Unexpected {}", fn_name);
+
+        HANDLER.with(|handler| match self.expected_reaction {
+            LintRuleReaction::Error => {
+                handler.struct_span_err(span, &message).emit();
+            }
+            LintRuleReaction::Warning => {
+                handler.struct_span_warn(span, &message).emit();
+            }
+            _ => {}
+        });
+    }
+
+    fn check_names(&self, obj_name: Option<&str>, fn_name: &str) {
+        if let Some(obj_name) = obj_name {
+            if self.pass_call_on_global_this && obj_name == GLOBAL_THIS_PROP {
+                return;
+            }
+
+            if !OBJ_NAMES.contains(&obj_name) {
+                return;
+            }
+        }
+
+        if FN_NAMES.contains(&fn_name) {
+            self.emit_report(fn_name);
+        }
+    }
+
+    fn check_member_expr(&self, member_expr: &MemberExpr) {
+        let MemberExpr { obj, prop, .. } = member_expr;
+
+        if let Expr::Ident(obj) = obj.as_ref() {
+            if obj.span.ctxt != self.top_level_ctxt {
+                return;
+            }
+
+            if self.top_level_declared_vars.contains(&obj.to_id()) {
+                return;
+            }
+
+            let obj_sym = Some(&*obj.sym);
+
+            match prop {
+                MemberProp::Ident(prop) => self.check_names(obj_sym, &*prop.sym),
+                MemberProp::Computed(comp) => {
+                    if let Expr::Lit(Lit::Str(lit_str)) = comp.expr.as_ref() {
+                        self.check_names(obj_sym, &*lit_str.value)
+                    }
+                }
+                _ => {}
+            }
+        }
+
+        // TODO: handle call alert on "this"
+    }
+}
+
+impl Visit for NoAlert {
+    noop_visit_type!();
+
+    fn visit_expr(&mut self, expr: &Expr) {
+        match expr {
+            Expr::Ident(id) => {
+                if id.span.ctxt != self.top_level_ctxt {
+                    return;
+                }
+
+                if self.top_level_declared_vars.contains(&id.to_id()) {
+                    return;
+                }
+
+                if self.call_expr_span.is_some() {
+                    self.check_names(None, &*id.sym);
+                }
+            }
+            Expr::Call(call_expr) => {
+                if self.call_expr_span.is_none() {
+                    self.call_expr_span = Some(call_expr.span);
```

Comment:
I think this will error for

```js
foo(() => alert("foo"));
```

as `call_expr_span` points `foo` call while vaildating `alert`.